dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderFactoryImpl.java

```diff
@@ -972,31 +976,18 @@ protected Folder findSystemFolder() throws DotDataException {
 
 	@SuppressWarnings("unchecked")
 	protected List<Folder> findFoldersByHost(Host host) throws DotHibernateException {
-		HibernateUtil dh = new HibernateUtil(Folder.class);
-		dh.setSQLQuery("SELECT {folder.*} from folder folder,identifier ident, inode folder_1_ where folder_1_.inode = folder.inode "
-			    + "and folder.identifier = ident.id and ident.host_inode = ? and ident.parent_path='/' order by lower(folder.title)");
-		dh.setParam(host.getIdentifier());
-		List<Folder> folderList=dh.list();
-		Collections.sort(folderList,new Comparator<Folder>() {
-		    public int compare(Folder o1, Folder o2) {
-		        return o1.getName().compareToIgnoreCase(o2.getName());
-		    }
-        });
+		List<Folder> folderList = getSubFolders(null, "/", host.getIdentifier(), null);
+		Collections.sort(folderList, (Folder folder1, Folder folder2) -> folder1.getName()
+				.compareToIgnoreCase(folder2.getName()));
 		return folderList;
 	}
 
 	@SuppressWarnings("unchecked")
 	protected List<Folder> findThemesByHost(Host host) throws DotHibernateException {
-		HibernateUtil dh = new HibernateUtil(Folder.class);
-		dh.setSQLQuery("SELECT {folder.*} from folder folder,identifier ident, inode folder_1_ where folder_1_.inode = folder.inode "
-			    + "and folder.identifier = ident.id and ident.host_inode = ? and ident.parent_path='/application/themes/' order by lower(folder.title)");
-		dh.setParam(host.getIdentifier());
-		List<Folder> folderList=dh.list();
-		Collections.sort(folderList,new Comparator<Folder>() {
-		    public int compare(Folder o1, Folder o2) {
-		        return o1.getName().compareToIgnoreCase(o2.getName());
-		    }
-        });
+		List<Folder> folderList = getSubFolders(null, "/application/themes/", host.getIdentifier(),

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'o1', 'o2' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)